src/odemis/driver/smaract.py

```diff
@@ -2931,6 +2955,15 @@ def _doReference(self, future, axes):
                         logging.warning("Axis %s not referenced after the end of referencing", a)
                         # TODO: Raise some error here
 
+                    # if referenced, move to the safe posiiton (if defined)
+                    if self._deactive_pos_after_ref is not None and self._deactive_pos_after_ref.get(a) is not None:
```

Comment:
`self._deactive_pos_after_ref.get(a) is not None` -> `a in self._deactive_pos_after_ref`